dbt/include/snowflake/macros/relations/dynamic_table/create.sql

```diff
@@ -1,8 +1,16 @@
 {% macro snowflake__get_create_dynamic_table_as_sql(relation, sql) -%}
 
+    {%- set dynamic_table = relation.from_config(config.model) -%}
```

Comment:
Using the `RelationConfig` allows us to easily incorporate input validation for `refresh_mode` and `initialize`. If we're doing that, we should pull all attribution from this class, hence updating `target_lag` and `snowflake_warehouse`.